### ISSUE TITLE
Support for `padded-blocks` corner cases

### ISSUE THREAD

**Issue by yahya-abou-imran** [prettier/prettier-eslint#799]
**Versions**:

- `prettier-eslint` version: ^15.0.1
- `node` version: v16.16.0
- `npm` (or `yarn`) version: 8.19.1

**Relevant code or config**

```javascript
const format = require("prettier-eslint");

const sourceCode = `
class User {

  constructor() {}

}

if (true) {

  console.log('okay')
  
}
`;

const options = {
  text: sourceCode,
  eslintConfig: {
    parserOptions: {
      ecmaVersion: 7,
    },
    rules: {
      "padded-blocks": ["error", {"classes": "always"}]
    },
  },
};

format(options).then(console.log)

```

Output:
```
class User {

  constructor() {}

}

if (true) {
  console.log("okay")
}
```

**Problem description:**

`prettier-eslint` is mentionned as a solution for the unsupported `padded-blocks` option [here](https://github.com/prettier/prettier/issues/2352#issuecomment-312429428).    
But it doesn’t seems to work for all cases, new lines in the `if` block are stripped by `prettier`.
The wanted output is :

```
class User {

  constructor() {}

}

if (true) {

  console.log("okay")

}
```

The reason is that `prettier` is formatting the `if` block, even if it’s not mark as error or warn by `eslint`.  
And after that, `eslint` does not complain, because nothing is enforced (except for classes).

**Comment by yahya-abou-imran** [comment]
As a notice: it could be seen as a more generic issue, that is: only run prettier for errors/warns.  
I actually wanted to open an issue with this kind of title.  
This problem could even be out-of-scope of this lib…

**Comment by qyurila** [comment]
> As a notice: it could be seen as a more generic issue, that is: only run prettier for errors/warns.

That's not possible considering how Prettier works. Prettier is made for formatting the whole code as it wants. If you want to get your desired output, you should enforce that rule by ESLint config file. e.g.

```ts
rules = {
  "padded-blocks": "error"  // default is "always"
}
```

...or just consider to don't using Prettier, if the above is not really what you want to do.
